nearcore/src/migrations.rs

```diff
@@ -478,6 +479,28 @@ pub fn migrate_24_to_25(path: &Path) {
     set_store_version(&store, 25);
 }
 
+/// Fix an issue with block ordinal (#5761)
+pub fn migrate_30_to_31(path: &Path, near_config: &NearConfig) {
+    let store = create_store(path);
+    if near_config.client_config.archive && &near_config.genesis.config.chain_id == "mainnet" {
+        let genesis_height = near_config.genesis.config.genesis_height;
+        let mut chain_store = ChainStore::new(store.clone(), genesis_height);
+        let head = chain_store.head().unwrap();
+        let mut store_update = BatchedStoreUpdate::new(&store, 10_000_000);
+        // we manually checked mainnet archival data and the first block where the discrepancy happened is `47443088`.
```

Comment:
did we verify every archival node that exists ? (as AFAIK this bug was not deterministic)

Reply:
We verified it on multiple of our archival nodes and they all have the same issue.